homeassistant/components/binary_sensor/xiaomi_aqara.py

```diff
@@ -187,6 +202,112 @@ def parse_data(self, data, raw_data):
             return True
 
 
+class XiaomiUnlockSensor(XiaomiBinarySensor):
+    """Representation of a XiaomiUnlockSensor."""
+
+    def __init__(self, config, devices, device, hass, xiaomi_hub):
+        """Initialize the XiaomiMotionSensor."""
+        self._hass = hass
+        self._no_motion_since = 0
+        self.sub_devices = {}
+        users = config[CONF_LOCKUIDS] if CONF_LOCKUIDS in config else []
+        users.append({
+            'uid': 'unsecu'
+        })
+        for user in users:
+            self.sub_devices[str(user['uid'])] = XiaomiUnlockSubSensor({
+                'sid': str(user['uid']),
+                'real_sid': device['sid'],
+                'model': 'motion',
+                'data': {
+                },
+                'raw_data': {
+                    'cmd': 'report'
+                }
+            }, hass, xiaomi_hub)
+            devices.append(self.sub_devices[str(user['uid'])])
+        XiaomiBinarySensor.__init__(self, device, 'Unlock Sensor', xiaomi_hub,
+                                    'status', 'motion')
+
+    def push_sub_devices(self, uid=None):
+        """Parse data sent by gateway."""
+        for key in self.sub_devices:
+            self.sub_devices[key].push_data({
+                'status':
+                    MOTION if uid is not None and uid == key else NO_MOTION
+            }, {
+                'sid': key,
+                'cmd': 'report',
+                'status':
+                    MOTION if uid is not None and uid == key else NO_MOTION
+            })
+
+    def parse_data(self, data, raw_data):
+        """Parse data sent by gateway."""
+        if raw_data['cmd'] in ['heartbeat', 'read_ack', 'read_rsp']:
```

Comment:
Do you really want to skip the heartbeat? This was/is a workaround for motion sensors.